Doc/c-api/frame.rst

```diff
@@ -41,6 +41,17 @@ See also :ref:`Reflection <reflection>`.
    .. versionadded:: 3.9
 
 
+.. c:function:: PyCodeObject* PyFrame_GetLocals(PyFrameObject *frame)
+
+   Get the *frame*'s ``f_locals`` attribute.
```

Comment:
```suggestion
   Get the *frame*'s ``f_locals`` attribute (:class:`dict`).
```

Would you mind to mention that this function **can** fail (return NULL and raises an exception)? It's uncommon that reading an object attribute can raise an exception. Users don't know Python internals and the complexity of PyFrame_FastToLocalsWithError().

Would it be possible to say something about modifications of this dict, if changes are expected or not to be reflected in the frame. From what I understood: "it depends" and PyFrame_LocalsToFast() **must** be called to make the update more likely.

Reply:
I think it can only fail because of memory error or failing to set item on locals dictionary if it is moneypatched with any other object, both are unlikely.

Reply:
Isn't it the default that an API function can raise and return NULL?
It makes sense to document those functions that cannot fail, as they are unusual.

Documenting almost all API functions with "this can fail..." just makes the docs harder to read, IMO.